x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -113,6 +113,13 @@ const getSavedObjectTypes = (
         monitoring_output_id: { type: 'keyword' },
         download_source_id: { type: 'keyword' },
         fleet_server_host_id: { type: 'keyword' },
+        agent_features: {
+          dynamic: false,
```

Comment:
@elastic/kibana-core would someone be able to review this SO change, the only thing of note is that I've set `dynamic: false` as we want to allow arbitrary values, not sure if there are any risks here?

Reply:
@rudolf has an open PR that cleans up saved object fields with mappings configured as `enabled: false` because it provides insufficient validation and allows indexing a string into these objects.

Allowing arbitrary values is risky regarding the goal of versioned APIs, particularly when supporting `find` operations. In the future, using `find` for any arbitrary field will make validation really hard in your versioned APIs.

Can you refactor to only index a known subset of searchable fields rather?

Reply:
to add to that, having unknown fields is also going to make your life hard w.r.t BWC. How would you handle a backward-compatible migration for something that's unknown?

Reply:
`dynamic: false` just means all the underlying fields won't be indexed. Unless I missed something in @rudolf's last work, this is still the recommended way of storing unindexed data?

@TinaHeiligers in term of BWC compliance, we need to be aware that schema != model. Having data being non-indexed doesn't really have an impact on BWC (or non-BWC).

Reply:
Ok I'll wait to see what Rudolf thinks too, thanks so much for the input, it was a slight red flag to me which is why I called it out  👍

Reply:
yeah these mappings looks fine to me :+1: and these arbitrary values will only ever be stored or retrieved, there won't be any find/search over them.

This API is tightly coupled to the persistence (which we don't recommend). Whatever is stored will be returned which could be a problem for BWC if ever any of the unknown keys in `agent_features` got transformed by a migration. But I suspect this is "by design" and fleet would never touch this data on the Kibana side.

Reply:
Oh, I just noticed these values come from the Kibana UI. So while the mappings still look perfect to me I think if we can, having the API validate the input would be preferable over just allowing any unknowns.